shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h

```diff
@@ -118,12 +121,14 @@ class FlutterPlatformViewLayerPool {
   size_t available_layer_index_ = 0;
   std::vector<std::shared_ptr<FlutterPlatformViewLayer>> layers_;
 
+  const std::shared_ptr<IOSSurfaceFactory> ios_surface_factory_;
+
   FML_DISALLOW_COPY_AND_ASSIGN(FlutterPlatformViewLayerPool);
 };
 
 class FlutterPlatformViewsController {
  public:
-  FlutterPlatformViewsController();
+  FlutterPlatformViewsController(std::shared_ptr<IOSSurfaceFactory> surface_factory);
```

Comment:
nit: `const std::shared_ptr<IOSurfaceFactory>&`  shared_ptr's are threadsafe so copying the objects incurs locking primitives which are system calls that slow down execution.

Reply:
This is interesting, I've been passing `shared_ptr<T>` by value rather than `const shared_ptr<T>&` as a convention. But clearly that has very little use, I will change it to `const shared_ptr<T>&` given that `FlutterPlatformViewsController` has no intention of sharing ownership of `surface_factory`. Also came across [this](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#r36-take-a-const-shared_ptrwidget-parameter-to-express-that-it-might-retain-a-reference-count-to-the-object-) which makes sense.

Reply:
Yea, the engine doesn't use const refs for passing arguments when it should.  I got yelled at in the past for passing std::functions by const reference.  We don't use shared_ptr's much but they are particularly important to pass by const reference because of the mutex inside them.  I've worked on projects that have been crippled by that overhead.